javascript/modules/client/src/client.ts

```diff
@@ -579,50 +593,152 @@ export class Client extends ClientCore implements IClient {
 
   //// PRIVATE METHODS METADATA
 
-  private _getDao(daoAddressOrEns: string): Promise<DaoDetails> {
-    // TODO: Implement actual fetch logic using subgraph.
-
-    if (!daoAddressOrEns) {
-      throw new Error("Invalid DAO address or ENS");
+  private async _getDao(daoAddressOrEns: string): Promise<DaoDetails | null> {
+    if (!isAddress(daoAddressOrEns)) {
+      // TODO
+      // ens resolve
+      throw new Error("Invalid address");
+    }
+    try {
+      await this.graphql.ensureOnline();
+      const client = this.graphql.getClient();
+      const { dao } = await client.request(QueryDao, {
+        address: daoAddressOrEns,
+      });
+      if (!dao) {
+        return null;
+      }
+      // const stringMetadata = await this.ipfs.fetchString(dao.metadata);
+      // TODO
+      // this is a temporal fix and should be changed by the line above
+      // but the current daos in subgraph dont have a proper metadata
+      const stringMetadata = await this.ipfs.fetchString(
+        "QmebY8BGzWBUyVqZFMaFkFmz3JrfaDoFP5orDqzJ1uiEkr",
+      );
+      const metadata = JSON.parse(stringMetadata);
```

Comment:
We already have a task for that, but still good to have

```suggestion
      // TODO: Parse and validate schema
      const metadata = JSON.parse(stringMetadata);
```